src/cfg.py

```diff
@@ -309,11 +309,19 @@ def __str__(self):
                           (self.entry, self.exit))
         b_id = self.ident() if self.entry is not None else "Unidentified"
         head = "Block {}\n[{}:{}]".format(b_id, entry, exit)
-        pred = "Predecessors: [{}]".format(", ".join(b.ident() for b in self.preds))
-        succ = "Successors: [{}]".format(", ".join(b.ident() for b in self.succs))
+        pred = "Predecessors: [{}]".format(", ".join(b.ident() for b in sorted(self.preds)))
+        succ = "Successors: [{}]".format(", ".join(b.ident() for b in sorted(self.succs)))
         unresolved = "\nHas unresolved jump." if self.has_unresolved_jump else ""
         return "\n".join([head, self._STR_SEP, pred, succ]) + unresolved
 
+    def __lt__(self, other):
+        """
+        Compare BasicBlocks based on their entry program counter values.
+        """
+        if self.entry is None or other.entry is None:
+            return True
```

Comment:
What's the rationale behind returning True in this case?

Reply:
I guess it could be false. I chose true so that `None` appears at the start of a sorted list.